scripts/GetClrDbg.sh

```diff
@@ -49,32 +48,63 @@ print_help()
     echo '  <version> can be "latest" or a version number such as 14.0.25109-preview-2865786'
 }
 
-# Sets __RuntimeID as given by 'dotnet --info'
+# Set the __RuntimeID by reading the contents of /etc/os-release. 
+# This logic is identical to the one used by C# extensions, make sure they are kept in sync.
 get_dotnet_runtime_id()
 {
-    # example output of dotnet --info
-    # .NET Command Line Tools (1.0.0-beta-002194)
-    #
-    # Product Information:
-    #  Version:     1.0.0-beta-002194
-    #  Commit Sha:  a28369bfe0
-    #
-    # Runtime Environment:
-    #  OS Name:     ubuntu
-    #  OS Version:  14.04
-    #  OS Platform: Linux
-    #  RID:         ubuntu.14.04-x64
-
-    # Get the output of dotnet --info
-    info="$(dotnet --info)"
-
-    # filter to the line that contains the RID. Output still contains spaces. Example: '         ubuntu.14.04-x64'
-    rid_line="$(echo "${info}" | awk 'BEGIN { FS=":" }{ if ( $1 ~ "RID" ) { print $2 } }')"
-
-    # trim whitespace from awk return
-    rid="$(echo -e "${rid_line}" | tr -d '[[:space:]]')"
-
-    __RuntimeID=$rid
+    # Sample content of /etc/os-release looks like this
+    # NAME="Ubuntu"
+    # VERSION="14.04.5 LTS, Trusty Tahr"
+    # ID=ubuntu
+    # ID_LIKE=debian
+    # PRETTY_NAME="Ubuntu 14.04.5 LTS"
+    # VERSION_ID="14.04"
+    # HOME_URL="http://www.ubuntu.com/"
+    # SUPPORT_URL="http://help.ubuntu.com/"
+    # BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
+
+    PlatformID=$(cat /etc/os-release | awk '{split($0,a,"="); if (a[1] == "ID") print a[2]}' | tr -d \" | tr -d '[[:space:]]')
+    PlatformVersionID=$(cat /etc/os-release | awk '{split($0,a,"="); if (a[1] == "VERSION_ID") print a[2]}' | tr -d \" | tr -d '[[:space:]]')
+
+    case $PlatformID in
+        ubuntu) 
+            if [[ "$PlatformVersionID" == 14* ]]; then
+                __RuntimeID=ubuntu.14.04-x64
+            elif [[ "$PlatformVersionID" == 16* ]]; then
+                __RuntimeID=ubuntu.16.04-x64
+            fi
+            ;;
+        centos)
+            __RuntimeID=centos.7-x64
+            ;;
+        fedora)
+            __RuntimeID=fedora.23-x64
+            ;;
+        opensuse)
+            __RuntimeID=opensuse.13.2-x64
+            ;;
+        rhel)
+            __RuntimeID=rhel.7-x64
+            ;;
+        debian)
+            __RuntimeID=debian.8-x64
+            ;;
+        ol)
+            __RuntimeID=centos.7-x64
+            ;;
+        elementaryOS)
+            if [[ "$PlatformVersionID" == 0.3* ]]; then
```

Comment:
Looks like it doesn't need to be.

---
In reply to: [77927758](https://github.com/Microsoft/MIEngine/pull/437#discussion_r77927758) [](ancestors = 77927758)